### PR TITLE
Update peer_tags.ini

### PR DESCRIPTION
**Pull request by majorgreys**
### What does this PR do?

Update to latest peer tag mapping.

### Describe how to test/QA your changes

Similar to https://github.com/DataDog/datadog-agent/pull/28230, we should:

- Enable both `apm_config.peer_service_aggregation` and `apm_config.compute_stats_by_span_kind`.
- Send `span.kind=span.kind` spans to the trace agent with `db.type:foo` set in the meta.
- Validate that the `peer.db.system:foo` tag is available on the client metrics

### CODE REVIEW

**Comment by knusbaum** [comment]
I don't see the `db.system` ones in the semantic core file:
https://github.com/DataDog/semantic-core/blob/main/generated/domains/apm/peer_tags/peer-tags-config.ini

Shouldn't this be an exact copy of that?

**Comment by majorgreys** [comment]
> I don't see the `db.system` ones in the semantic core file

We are using a new source of truth that is shared with the backend

**Review by urseberry — APPROVED** [review]

File: releasenotes/notes/apm-update-peer-tags-c4ca55ac8e190114.yaml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+enhancements:
```
Comment (ichinaski) [inline releasenotes/notes/apm-update-peer-tags-c4ca55ac8e190114.yaml:1]: Not sure release notes are needed for this PR? If not just add the label `no-changelog`

File: pkg/trace/config/peer_tags_test.go [hunk 2]
```diff
@@ -43,60 +42,6 @@ func TestPreparePeerTags(t *testing.T) {
...
-}
-
-func toSet(list []string) map[string]bool {
-	result := make(map[string]bool)
-	for _, l := range list {
-		result[l] = true
-	}
-	return result
+	// Simple test to ensure peer tags are loaded
+	assert.Contains(t, basePeerTags, "db.name")
```
Comment (ichinaski) [inline pkg/trace/config/peer_tags_test.go:46]: Why do we remove the other checks?

Comment (majorgreys) [reply pkg/trace/config/peer_tags_test.go:46]: https://github.com/DataDog/datadog-agent/pull/28230/files#diff-689864b3834218d5bd46597d505f9bba897c143d718bfe917dac912482b017bf

Comment (majorgreys) [reply pkg/trace/config/peer_tags_test.go:46]: This unit test should not be exhaustive but rather that `basePeerTags` is non-empty and has a valid entry. We then do not have to keep this expected list updated as the ini file is updated.

File: releasenotes/notes/apm-update-peer-tags-c4ca55ac8e190114.yaml [hunk 3]
```diff
@@ -0,0 +1,3 @@
...
+  - |
+    Updates peer tags for ``peer.db.system``.
```
Comment (ichinaski) [inline releasenotes/notes/apm-update-peer-tags-c4ca55ac8e190114.yaml:3]:
```suggestion
    APM: Updates peer tags for ``peer.db.system``.
```

File: releasenotes/notes/apm-update-peer-tags-c4ca55ac8e190114.yaml (hunk 1 above)
Comment (ichinaski) [reply releasenotes/notes/apm-update-peer-tags-c4ca55ac8e190114.yaml:1]: Makes sense. Made a small suggestion but otherwise LGTM.

**Review by ichinaski — APPROVED** [review]

**Comment by ichinaski** [comment]
Re: QA, I managed to successfully test the feature, but the `span.kind` had to be set to `client`. Any other value would result on the peer tags not being available on the metrics.

**Pull request merged**